src/cleanfigure.m

```diff
@@ -114,16 +114,32 @@ function cleanfigure(varargin)
           movePointsCloser(meta, h);
           % Don't be too precise.
           coarsenLine(meta, h, minimumPointsDistance);
+
       elseif strcmp(type, 'text')
-          % Check if text is inside bounds by checking if the Extent rectangle
-          % and the axes box overlap.
-          xlim = get(meta.gca, 'XLim');
-          ylim = get(meta.gca, 'YLim');
+          % Check if text is inside bounds by checking if the position is inside
+          % the x, y and z limits. This works for both 2D and 3D plots.
+          x_lim = get(meta.gca, 'XLim');
+          y_lim = get(meta.gca, 'YLim');
+          z_lim = get(meta.gca, 'ZLim');
+          axLim = [x_lim; y_lim; z_lim];
+
+          pos = get(h, 'Position');
+          bPosInsideLim = ( pos' >= axLim(:,1) ) & ( pos' <= axLim(:,2) );
+
+          % In 2D plots the extent of the textbox is available and also
+          % considered to keep the textbox, if it is partially inside the axis
+          % limits.
           extent = get(h, 'Extent');
-          extent(3:4) = extent(1:2) + extent(3:4);
-          overlap = xlim(1) < extent(3) && xlim(2) > extent(1) ...
-                 && ylim(1) < extent(4) && ylim(2) > extent(2);
-          if ~overlap
+          if all(~isnan(extent))
```

Comment:
As I was mentioning, I think we filter out boxes outside the axes in order to avoid catching the xyz labels, titles and tick labels created ad hoc by specific plots as texts. If we were to convert those as texts, then other axes specific routines would re-draw a second time resulting in duplication. 

Now, things changed quite a bit with HG2 and it may indeed not consitute a problem anymore, but also in R2014a I am not sure this is a big issue and we should try to come up with an exact example on why we want ot filter out text boxes (maybe we weren't able to handle annotations lying on the figure, as another reason). Otherwise, I would suggest to simply kill that check and try to draw everything.